pentaho-blueprint-collection-utils/src/main/java/org/pentaho/osgi/blueprint/collection/utils/ServiceMap.java

```diff
@@ -28,11 +28,13 @@
  */
 public class ServiceMap<T> {
 
-  private final Map<String, MapItem<T>> map = new HashMap<String, MapItem<T>>();
+  private final Map<String, MapItem<T>> map = new ConcurrentHashMap<>();
   public static final String SERVICE_KEY_PROPERTY = "serviceMapKey";
 
   public T getItem( String key ) {
-    return map.get( key ) != null ? map.get( key ).getItem() : null;
+    synchronized ( map ) {
```

Comment:
Since using `synchronized` remove the benefits of using a `ConcurrentHashMap` over `HashMap`, I'd suggest removing all `synchronized ( map )` blocks in this class. 
`ConcurrentHashMap` manages this by itself.